src/effects/backends/builtin/echoeffect.cpp

```diff
@@ -118,6 +118,14 @@ void EchoEffect::loadEngineEffectParameters(
     m_pTripletParameter = parameters.value("triplet");
 }
 
+float averageSampleEnergy(mixxx::SampleBuffer const& delay_buffer) {
```

Comment:
don't take a reference to an owning type if you can take the non-owning counterpart directly.
```suggestion
float averageSampleEnergy(std::span<const CSAMPLE> delay_buffer) {
```